File: casestudy/module2/Module2_WBE/src/main/java/com/codegym/service/nhanvien/NhanVienService.java
package com.codegym.service.nhanvien;

import com.codegym.model.nhanvien.NhanVien;

import java.util.List;

public interface NhanVienService {

    List<NhanVien> findAll();

    NhanVien findById(Integer id);

    void save(NhanVien nhanVien);

    void remove(Integer id);
}

File: casestudy/module3/furama-resort/src/app/Service/nhanvien.service.ts
import {Injectable} from '@angular/core';
import {NhanvienModel, listNhanVien} from '../model/nhanvien.model';

@Injectable({
  providedIn: 'root'
})
export class NhanvienService {

  listNhanVien: NhanvienModel[];

  constructor() {
    this.listNhanVien = listNhanVien;
  }

  findAll(): NhanvienModel[] {
    return this.listNhanVien;
  }

  findById(id: string): NhanvienModel {
    return this.listNhanVien.find(khachHang => khachHang.idNhanVien === id);
  }

  save(nhanvien: NhanvienModel): void {
    listNhanVien.push(nhanvien);
  }
}
